subsys/net/ip/net_context.c

```diff
@@ -1814,29 +1814,27 @@ NET_CONN_CB(tcp_syn_rcvd)
 		}
 
 		/* Swap the newly-created TCP states with the one that
-		 * was used to establish this connection. The new TCP
+		 * was used to establish this connection. The old TCP
 		 * must be listening to accept other connections.
 		 */
-		tmp_tcp = new_context->tcp;
-		tmp_tcp->accept_cb = tcp->accept_cb;
-		tcp->accept_cb = NULL;
-		new_context->tcp = tcp;
+		new_tcp = new_context->tcp;
+		new_tcp->accept_cb = tcp->accept_cb;
```

Comment:
Hmm, but why do we copy accept_cb here? It's more than unlikely that newly accepted ctx will be turned into listen one and used for accepting connections again.

Reply:
Few lines below we call this newly accepted cb, so need to set it.

Reply:
So, we should call it via the original, listening socket, and for newly accepted context, it should be set to NULL.

Reply:
Indeed, you are absolutely right. I will send a new version.